internal/internal.go

```diff
@@ -718,3 +758,59 @@ type internalError struct {
 func (e *internalError) Error() string {
 	return e.Message
 }
+
+func (p *envoyExtAuthzV2Wrapper) Check(ctx context.Context, req *ext_authz_v2.CheckRequest) (*ext_authz_v2.CheckResponse, error) {
+	var stop func() *rpc_status.Status
+	respV3, stop, err := p.v3.check(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	respV2 := v2Response(respV3)
+	if code := stop(); code != nil {
+		respV2.Status = code
+	}
+	return respV2, nil
```

Comment:
I assume the conversion is relatively quick and accounts for a small portion (e.g., 1%) of the check response time...is that right? Since we're not including it in the decision log metrics, I just want to make sure it's negligible and can be ignored.

Reply:
It was my intention to _include it_ in the decision log time, that's where the awkward `stop` handling came from. It's converted in L769 right before stopping the time in L770. I might be mislead about how it works...? 😄

Reply:
Whoops. I didn't look closely enough at what `stop()` was doing. The metric tracking looks fine then.

Reply:
However, we'll want to call `stop()` regardless of whether an error occurred--we want a decision log emitted in the case of errors.

Reply:
Yep, that's an oversight. We'll want `stop` to happen for the L756 case, too, indeed.

Reply:
I've fixed it, and duplicated a few of the unit tests for v2, to cover this logic.